sdk/tables/azure-data-tables/inc/azure/data/tables/tables_sas_builder.hpp

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "azure/data/tables/account_sas_builder.hpp"
+#include "azure/data/tables/credentials/shared_key_credential.hpp"
+
+#include <azure/core/datetime.hpp>
+#include <azure/core/nullable.hpp>
+
+#include <string>
+#include <type_traits>
+
+namespace Azure { namespace Data { namespace Tables { namespace Sas {
+  /**
+   * @brief Defines the protocols permitted for Storage requests made with a shared
+   * access signature.
+   */
+  enum class TablesSasProtocol
+  {
+    /**
+     * @brief No protocol has been specified. If no value is specified,
+     * the service will default to HttpsAndHttp.
+     */
+    None = 0,
+
+    /**
+     * @brief Only requests issued over HTTPS or HTTP will be permitted.
+     */
+    HttpsAndHttp = 1,

```

Comment:
Is there actually a real-world scenario where we allow HTTP access to the Tables service?

I thought we had a hard and fast rule that no azure services allowed HTTP access.

Reply:
probably not but it is defined as such in the documentation ,other storage service implementations and also in go and .net , thus ..